src/error.rs

```diff
@@ -193,6 +193,98 @@ impl Error {
     pub(crate) fn from_resolve_error(error: trust_dns_resolver::error::ResolveError) -> Self {
         ErrorKind::DnsResolveError { message: error.to_string() }.into()
     }
+
```

Comment:
These methods were moved from being `impl ErrorKind` to just being on `impl Error` since the `Deref` implementation is gone now. They're all internal anyways so there isn't any impact on the public API.